sky/clouds/azure.py

```diff
@@ -651,6 +651,10 @@ def query_status(cls, name: str, tag_filters: Dict[str, str],
         assert stdout.strip(), f'No status returned for {name!r}'
 
         original_statuses_list = json.loads(stdout.strip())
+        if not original_statuses_list:
```

Comment:
Should we check those inside the `if not isinstance(original_statuses_list, list):` if branch in L658? Seems like this check contains two possibilities:
* an empty string
* an empty list
the latter should be processed in following logic?

Reply:
I guess it should be fine as we are returning immediately anyway. We should check empty with `not`. Reference: https://google.github.io/styleguide/pyguide.html#2144-decision
wdyt?